pkg/provision/workspace/deployment_provisioning.go

```diff
@@ -13,12 +13,12 @@
 // limitations under the License.
 //
 
-package metrics
+package workspace
```

Comment:
This file shouldn't be in `pkg/provision/workspace`, as you mention in the description. If we're seeing an import cycle it points at a general problem -- why does the metrics package depend on pkg/provision and vice versa?
* If metrics depends on `pkg/provision/workspace`, why does metrics care about how a deployment is created?
* If `pkg/provision/workspace` depends on metrics, why should it care about how the controller itself reports metrics?

IMO metrics should be only imported in the `controllers/workspace` package, as it's a library for that package -- importing metrics in a `pkg/provision` package ties the two packages together in a way we don't want.

One way to avoid this problem in this instance is to not use `DeploymentProvisioningStatus` -- it's an unnecessary object passed in to only get the Info() message (a string), which is the actual data we're interested in.